packages/confusing-browser-globals/index.js

```diff
@@ -7,63 +7,72 @@
 
 'use strict';
 
-module.exports = [
-  'addEventListener',
-  'blur',
-  'close',
-  'closed',
-  'confirm',
-  'defaultStatus',
-  'defaultstatus',
-  'event',
-  'external',
-  'find',
-  'focus',
-  'frameElement',
-  'frames',
-  'history',
-  'innerHeight',
-  'innerWidth',
-  'length',
-  'location',
-  'locationbar',
-  'menubar',
-  'moveBy',
-  'moveTo',
-  'name',
-  'onblur',
-  'onerror',
-  'onfocus',
-  'onload',
-  'onresize',
-  'onunload',
-  'open',
-  'opener',
-  'opera',
-  'outerHeight',
-  'outerWidth',
-  'pageXOffset',
-  'pageYOffset',
-  'parent',
-  'print',
-  'removeEventListener',
-  'resizeBy',
-  'resizeTo',
-  'screen',
-  'screenLeft',
-  'screenTop',
-  'screenX',
-  'screenY',
-  'scroll',
-  'scrollbars',
-  'scrollBy',
-  'scrollTo',
-  'scrollX',
-  'scrollY',
-  'self',
-  'status',
-  'statusbar',
-  'stop',
-  'toolbar',
-  'top',
+const restrictedGlobals = [
+  ['addEventListener', 'document'],
+  ['blur', 'window'],
+  ['close', 'window'],
+  ['closed', 'window'],
+  ['confirm', 'window'],
+  ['defaultStatus', 'none'],
+  ['defaultstatus', 'none'],
+  ['event', 'none'],
+  ['external', 'none'],
+  ['find', 'none'],
+  ['focus', 'window'],
+  ['frameElement', 'window'],
+  ['frames', 'window'],
+  ['history', 'window'],
+  ['innerHeight', 'window'],
+  ['innerWidth', 'window'],
+  ['length', 'window'],
+  ['location', 'window'],
+  ['locationbar', 'window'],
+  ['menubar', 'window'],
+  ['moveBy', 'window'],
+  ['moveTo', 'window'],
+  ['name', 'window'],
+  ['onblur', 'window'],
+  ['onerror', 'window'],
+  ['onfocus', 'window'],
+  ['onload', 'window'],
+  ['onresize', 'window'],
+  ['onunload', 'window'],
+  ['open', 'window'],
+  ['opener', 'window'],
+  ['opera', 'window'],
+  ['outerHeight', 'window'],
+  ['outerWidth', 'window'],
+  ['pageXOffset', 'window'],
+  ['pageYOffset', 'window'],
+  ['parent', 'window'],
+  ['print', 'window'],
+  ['removeEventListener', 'document'],
+  ['resizeBy', 'window'],
+  ['resizeTo', 'window'],
+  ['screen', 'window'],
+  ['screenLeft', 'window'],
+  ['screenTop', 'window'],
+  ['screenX', 'window'],
+  ['screenY', 'window'],
+  ['scroll', 'window'],
+  ['scrollbars', 'window'],
+  ['scrollBy', 'window'],
+  ['scrollTo', 'window'],
+  ['scrollX', 'window'],
+  ['scrollY', 'window'],
+  ['self', 'WorkerGlobalScope'],
```

Comment:
This PR would be useful! It has my 👍 .

`self` is a property on the WorkerGlobalScope _instance_ (it isn't static), so `WorkerGlobalScope.self` actually returns `undefined`.

A couple of alternatives to using `self` are: `globalThis` (older browsers not supported) and `this` (definitely not recommended). Personally, I believe `self` is better than either of these.